atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepQueueComponents.java

```diff
@@ -0,0 +1,118 @@
+import com.palantir.atlasdb.schema.TargetedSweepSchema;
+import com.palantir.atlasdb.sweep.metrics.TargetedSweepMetrics;
+import com.palantir.atlasdb.sweep.queue.NumberOfShardsProvider.MismatchBehaviour;
+import com.palantir.atlasdb.sweep.queue.SweepQueueReader.ReadBatchingRuntimeContext;
+import com.palantir.atlasdb.sweep.queue.clear.DefaultTableClearer;
+import com.palantir.atlasdb.table.description.Schemas;
+import com.palantir.atlasdb.transaction.impl.TimelockTimestampServiceAdapter;
+import com.palantir.atlasdb.transaction.service.TransactionService;
+import com.palantir.atlasdb.transaction.service.TransactionServices;
+import com.palantir.lock.v2.TimelockService;
+import java.time.Duration;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface SweepQueueComponents {
+    SweepQueueReader reader();
+
+    SweepQueueWriter writer();
+
+    SweepQueueDeleter deleter();
+
+    SweepQueueCleaner cleaner();
+
+    ShardProgress shardProgress();
+
+    SweepableCells sweepableCells();
+
+    SweepableTimestamps sweepableTimestamps();
+
+    NumberOfShardsProvider numberOfShardsProvider();
+
+    static SweepQueueComponents create(
+            TargetedSweepMetrics metrics,
+            KeyValueService kvs,
+            TimelockService timelock,
+            Supplier<Integer> shardsConfig,
+            ReadBatchingRuntimeContext readBatchingRuntimeContext,
+            TargetedSweepFollower follower,
+            MismatchBehaviour mismatchBehaviourForShards) {
+        // It is OK that the transaction service is different from the one used by the transaction manager,
+        // as transaction services must not hold any local state in them that would affect correctness.
+        TransactionService transaction =
+                TransactionServices.createRaw(kvs, new TimelockTimestampServiceAdapter(timelock), false);
+        return create(
+                metrics,
+                kvs,
+                timelock,
+                shardsConfig,
+                transaction,
+                readBatchingRuntimeContext,
+                _unused -> Optional.empty(),
+                follower,
+                mismatchBehaviourForShards);
+    }
+
+    static SweepQueueComponents create(
+            TargetedSweepMetrics metrics,
+            KeyValueService kvs,
+            TimelockService timelock,
+            Supplier<Integer> shardsConfig,
+            TransactionService transaction,
+            ReadBatchingRuntimeContext readBatchingRuntimeContext,
+            Function<TableReference, Optional<LogSafety>> tablesToTrackDeletions,
+            TargetedSweepFollower follower,
+            MismatchBehaviour mismatchBehaviourForShards) {
+        Schemas.createTablesAndIndexes(TargetedSweepSchema.INSTANCE.getLatestSchema(), kvs);
+        ShardProgress shardProgress = new ShardProgress(kvs);
```

Comment:
An argument can be made for injecting this separately and scoping the SweepQueueComponents to just the reader/writer e.t.c. 

I chose to add the tables to the components, since, well they're components for both ASTS and existing sweep, and it's what the old code did, but not strongly tied to that.